pkg/solana/logpoller/log_poller.go

```diff
@@ -210,6 +225,43 @@ func (lp *Service) UnregisterFilter(ctx context.Context, name string) error {
 	return lp.filters.UnregisterFilter(ctx, name)
 }
 
+// Replay submits a new replay request. If there was already a new replay request
+// submitted since the last replay completed, it will be updated to the earlier of the
+// two requested fromBlock's. The expectation is that, on the next timer tick of the
+// LogPoller run loop it will backfill all filters starting from fromBlock. If there
+// are new filters in the backfill queue, with an earlier StartingBlock, then they
+// will get backfilled from there instead.
+func (lp *Service) Replay(fromBlock int64) error {
+	lp.replay.mut.Lock()
+	defer lp.replay.mut.Unlock()
```

Comment:
I'm not usually a fan of using locks in this way. I tend to think getter and setter methods are safer and consolidates lock access instead of spreading them around a larger surface area. However, if it works for now it's good by me.